makes/funcs.mk

```diff
@@ -1,5 +1,14 @@
 ### simple utility/shortcut functions ###
 
+# Converts to upper case
+define uppercase
+$(shell echo '$1' | tr 'a-z' 'A-Z')
```

Comment:
my "one thing to consider" wasn't posted somehow, I wanted to mention that we previously had toupper make internal implementation:

```make
# (string)
TO_LOWER = $(subst A,a,$(subst B,b,$(subst C,c,$(subst D,d,$(subst E,e,$(subst F,f,$(subst G,g,$(subst H,h,$(subst I,i,$(subst J,j,$(subst K,k,$(subst L,l,$(subst M,m,$(subst N,n,$(subst O,o,$(subst P,p,$(subst Q,q,$(subst R,r,$(subst S,s,$(subst T,t,$(subst U,u,$(subst V,v,$(subst W,w,$(subst X,x,$(subst Y,y,$(subst Z,z,$1))))))))))))))))))))))))))

# (string)
TO_UPPER = $(subst a,A,$(subst b,B,$(subst c,C,$(subst d,D,$(subst e,E,$(subst f,F,$(subst g,G,$(subst h,H,$(subst i,I,$(subst j,J,$(subst k,K,$(subst l,L,$(subst m,M,$(subst n,N,$(subst o,O,$(subst p,P,$(subst q,Q,$(subst r,R,$(subst s,S,$(subst t,T,$(subst u,U,$(subst v,V,$(subst w,W,$(subst x,X,$(subst y,Y,$(subst z,Z,$1))))))))))))))))))))))))))
```
I know it's ugly but maybe it was done not to depend on external tools (potential Darwin incompatibility? I honestly don't remember...)

Reply:
Hmm, that nested thing :) , ok I've updated to your suggestion, but used `lowercase` for shortcut names because that convention is already used e.g. `my-dir`